### DISCUSSION TITLE
Custom Field for power port doesn't show up until its associated with a device?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#10293] [Q&A]
Maybe I'm not doing things in the correct sequence, maybe my issue is expected behavior. Hopefully someone can set me straight.

We want to track BTU output for all of our power supplies. When creating a device type and you create the power ports for that device type you are able to set the maximum draw and allocated draw.

We have created a custom field called 'BTU' and assigned it to the 'DCIM > power port' model.

However, we notice that when creating power ports on device types, and module types, this custom field is not present. It only becomes present when we go to actually create a device. Then when we click on the power ports we are able to see the custom field.

Why does the custom field only show up on devices and not device types or module types?

**Comment by candlerb** [comment]
Device types have a "[power port template](https://github.com/netbox-community/netbox/blob/v3.3.2/netbox/dcim/models/device_component_templates.py#L208-L228)" not a "power port" - and as far as I can see, you can't add custom fields on those.

**Comment by BYT3M3** [comment]
I submitted a feature request for this #17448

**Reply by ghost** [reply]
been asked for for years.
https://github.com/netbox-community/netbox/issues/10754
https://github.com/netbox-community/netbox/issues/10324
https://github.com/netbox-community/netbox/issues/9243
